sklearn/metrics/pairwise.py

```diff
@@ -1590,7 +1605,10 @@ def _parallel_pairwise(X, Y, func, n_jobs, **kwds):
 
 def _pairwise_callable(X, Y, metric, force_all_finite=True, **kwds):
     """Handle the callable case for pairwise_{distances,kernels}."""
-    X, Y = check_pairwise_arrays(X, Y, force_all_finite=force_all_finite)
+    check_length_only = kwds.pop("check_length_only", False)
```

Comment:
We should avoid using `kwds` if we can.

Here you can add `check_length_only` as a parameter in the signature.
It also means that we want to modify the docstring of `pairwise_distances` and propagate the parameter.

We will need to update the docstring of `pairwise_distances` then.

Reply:
yes, my [first comment[(https://github.com/scikit-learn/scikit-learn/pull/24674#issuecomment-1279824862) in PR was inspired from this question only.